python/mxnet/gluon/block.py

```diff
@@ -833,6 +848,12 @@ def _deferred_infer_shape(self, *args):
     def _call_cached_op(self, *args):
         if self._cached_op is None:
             self._build_cache(*args)
+        assert self._cached_op, "cached op is not None"
+        if self._callback:
+            self._cached_op._register_op_hook(self._callback, self._monitor_all)
+            if len(self._flags) >= 2 and self._flags[1]:
+                warnings.warn("Callback is not supported when static_shape=True "
```

Comment:
Why's this? If it's not supported, do not register the callback?

Reply:
else say something like experimental for static_shape=True if you are not confident, but it might work

Reply:
Yes I will say experimental, good point thanks. The reason this is experimental is because for certain code path the cached op directly calls engine push instead of calling InvokeOp. adding the callbacks in such cases was much more difficult since we just had the opr and the engine vars not the ndarray references. Please see line: https://github.com/apache/incubator-mxnet/blob/master/src/imperative/cached_op.cc#L689

For this case, it is not supported.